Source/WebCore/workers/service/server/SWServerWorker.cpp

```diff
@@ -324,6 +324,11 @@ void SWServerWorker::setHasPendingEvents(bool hasPendingEvents)
     registration->tryActivate();
 }
 
+bool SWServerWorker::isIdle(Seconds idleTime) const
+{
+    return !m_hasPendingEvents && (MonotonicTime::now() - m_lastNeedRunningTime) > idleTime;
```

Comment:
nit: could also use `ApproximateTime` instead of `MonotonicTime` in this patch given the durations involved.